assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import ResizeObserver from 'resize-observer-polyfill';
+
+/**
+ * WordPress dependencies
+ */
+import { useLayoutEffect, useRef, useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { LeftArrow, RightArrow, GridView } from '../../button';
+
+const PAGE_WIDTH = 72;
+const PAGE_HEIGHT = 128;
+
+const Wrapper = styled.div`
+	position: relative;
+	display: grid;
+	grid: "left-navigation carousel right-navigation" auto / 53px 1fr 53px;
+	height: 100%;
+	width: 100%;
+	color:  ${ ( { theme } ) => theme.colors.fg.v1 };
+`;
+
+const Area = styled.div`
+	grid-area: ${ ( { area } ) => area };
+	height: 100%;
+	width: 100%;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	flex-direction: column;
+	padding: 16px 0 24px;
+`;
+
+const List = styled( Area )`
+	flex-direction: row;
+	align-items: flex-start;
+	justify-content: ${ ( { hasHorizontalOverflow } ) => hasHorizontalOverflow ? 'flex-start' : 'center' };
+	height: 100%;
+	overflow-x: ${ ( { hasHorizontalOverflow } ) => hasHorizontalOverflow ? 'scroll' : 'hidden' };
+`;
+
+const Page = styled.button`
+	padding: 0;
+	margin: 0 5px;
+	border: 3px solid ${ ( { isActive, theme } ) => isActive ? theme.colors.selection : theme.colors.bg.v1 };
+	height: ${ PAGE_HEIGHT }px;
+	width: ${ PAGE_WIDTH }px;
+	background-color: ${ ( { isActive, theme } ) => isActive ? theme.colors.fg.v1 : theme.colors.mg.v1 };
+	flex: none;
+`;
+
+const GridViewButton = styled( GridView )`
+	position: absolute;
+	bottom: 24px;
+`;
+
+function Carousel() {
+	const { state: { pages, currentPageIndex, currentPageId }, actions: { setCurrentPage } } = useStory();
+	const [ hasHorizontalOverflow, setHasHorizontalOverflow ] = useState( false );
+	const listRef = useRef();
+	const pageRefs = useRef( [] );
+
+	useLayoutEffect( () => {
+		const observer = new ResizeObserver( ( entries ) => {
+			for ( const entry of entries ) {
+				const offsetWidth = entry.contentBoxSize ? entry.contentBoxSize.inlineSize : entry.contentRect.width;
+				setHasHorizontalOverflow( listRef.current.scrollWidth > offsetWidth );
+			}
+		} );
+
+		observer.observe( listRef.current );
+
+		return () => observer.disconnect();
+	}, [ pages.length ] );
+
+	useLayoutEffect( () => {
+		if ( hasHorizontalOverflow ) {
+			const currentPageRef = pageRefs.current[ currentPageId ];
+
+			currentPageRef.scrollIntoView( {
```

Comment:
Looks like in Safari would need checking if it exists -- editor not loading and getting this in Safari:
<img width="1348" alt="Screenshot 2020-01-17 at 14 29 45" src="https://user-images.githubusercontent.com/3294597/72615828-d81c3a80-3935-11ea-862e-1a988642d326.png">

Working ok for you?

Reply:
Haven't tested in Safari. But sounds like I need a workaround there or just do nothing.

Reply:
Wow browser support for this kind of scrolling really isn't good.

Reply:
Ended up doing the scrollIntoView in this case, as it's more a nice-to-have. For the `scrollBy` part I added a fallback.

Reply:
Hmm still not working for me in Safari, it looks like the issue is `currentPageRef` being `undefined`. Does it work for you on Safari?

Reply:
Works like a charm for me in Safari 13. The refs are always there. Weird 🤔

Reply:
Hmm, I'm also using Safari 13. No clue why it's breaking 😕Tried clearing cache and ran the installation commands to be sure everything exists and is built properly, no luck so far.

@spacedmonkey If you have a moment, would you mind testing if the PR works for you in Safari?

Reply:
Added a safeguard in f41109cb403e37164b9fcc7303177ffc64157a1b in the meantime.

Reply:
Confirmed that it does not break in Safari anymore 👍